### DISCUSSION TITLE
LAB 02.04- CARD TRICK

### DISCUSSION THREAD

**Discussion by saimonproyects** [rramosp/ai4eng.v1#163] [02 - PYTHON]
Buenas noches, podrían colaborarme por favor con la siguiente duda. Mi código funciona perfectamente pero la plataforma no me reconoce la respuesta. @rramosp   @rramosp 

 
  

> def run(c, n, n_heaps=3):

      if n_heaps == 3:
  
        for i in range(n_heaps):
          h= make_heaps(c)
          c=collect_heaps(h,n)
        an= c.index(n)
      
      else:
        for i in range(n_heaps):
          h= make_heaps(c)
          c=collect_heaps(h,n)
        an = c.index(n)
      return an

y la plataforma me arroja el siguiente error :

    
![Captura de Pantalla 2022-07-12 a la(s) 11 10 48 p  m](https://user-images.githubusercontent.com/87740310/178648960-1ab1f31f-6fbc-451c-9a2c-4f778b5487e2.png)

Sin embargo, el AssertionError se levanta cuando en los argumentos de la función se asignan valores indebidos. Es como si la plataforma trata de evaluar la función en valores que siempre levantan el error.

El código funciona correctamente ejecutado manualmente: (modifique el return de la función para poder comprobar visualmente) 
![Captura de Pantalla 2022-07-12 a la(s) 11 23 04 p  m](https://user-images.githubusercontent.com/87740310/178650273-49448fea-8028-4b07-a03b-34ac4da7f6a3.png)

**Comment by Felipe-RA** [comment]
Saludos saimon

Incluye por favor tu código para las funciones `make_heaps()` y `collect_heaps()`

También te invito a cuestionarte si es realmente necesario la estructura de `if` con la que divides tu función

¡Quedamos a la espera!
